### PR TITLE
nerpa_controller: handle P4 table default action 

### PR DESCRIPTION
**Pull request by debnil**
This PR enables the modification of a table's default action.

- If a P4 table does not have a `const default_action`, we now generate an additional DDlog output relation that enables changing a table's default action (via `p4info2ddlog`).
- `nerpa_controller` and `p4ext` now successfully process those output relations and change the default action.
- `snvs.p4` and `snvs.dl` are modified, so that tables not intended to have a modifiable default action are now explicitly annotated as such.

### CODE REVIEW

**Comment by debnil** [comment]
@mbudiu-vmw @blp: Let me know if the generated relation to modify the default action looks as expected.

File: nerpa_controller/src/lib.rs [hunk 1]
```diff
@@ -557,14 +558,33 @@ impl SwitchClient {
                             }
                         }
 
+                        // Check if the update should be to the default action.
+                        // When using P4Runtime, the default action can be set by a Write RPC with a
+                        //`TableEntry message for which the `FieldMatch` field is empty.
```
Comment (mbudiu-vmw) [inline nerpa_controller/src/lib.rs:563]: missing backtick in comment

File: nerpa_controller/src/lib.rs [hunk 2]
```diff
@@ -557,14 +558,33 @@ impl SwitchClient {
...
+                        //
+                        // We assume that a relevant DDlog relation's name, lowercased, includes "defaultaction".
```
Comment (mbudiu-vmw) [inline nerpa_controller/src/lib.rs:565]: This kind of stuff should be also documented in a higher-level design document.

File: p4ext/src/lib.rs [hunk 3]
```diff
@@ -1612,18 +1619,21 @@ pub fn get_pipeline_config(
 /// * `table_action` - the action to execute on match.
 /// * `field_matches` - values to match on.
 /// * `priority` - used to order entries.
+/// * `is_default_action` - used to modify the default action.
```
Comment (mbudiu-vmw) [inline p4ext/src/lib.rs:1622]: if true we are updating the default action

**Review by mbudiu-vmw — APPROVED** [review]

**Pull request merged**